packages/components/src/Button/Button.js

```diff
@@ -11,14 +12,19 @@ function Button(props, forwardedRef) {
 		className,
 		currentColor,
 		icon,
-		isActive = false,
+		isActive: isActiveProp = false,
 		isControl = false,
 		isSubtle = false,
 		size = 'medium',
 		variant = 'secondary',
 		...otherProps
 	} = useContextSystem(props, 'Button');
 
+	const { buttonGroup } = useButtonGroupContext();
+	const isButtonGroupActive =
+		buttonGroup?.state && buttonGroup?.state === otherProps.value;
```

Comment:
Why the `&&`? Is `buttonGroup?.state === otherProps.value` not sufficient? I think they're _slightly_ different meanings but is the duplicated conditional get necessary?

Reply:
>  Is buttonGroup?.state === otherProps.value not sufficient?

It may be! I think I had issues in the beginning, but they may be unrelated. Checking now.

Thank you! ❤️

Reply:
Update: Oh! I think it should be:

```jsx
	const isButtonGroupActive = buttonGroup && buttonGroup?.state === otherProps.value;
```

It **does** make a difference.

Here's where it's tricky.

`buttonGroup` may not exist. It only exists if the `Button` is used within `ButtonGroup`.
Cool.

Second part... `value` (initial value) may not be defined.

Which means that `buttonGroup?.state === otherProps.value` would evaluate to `true` for regular `Button` that live outside of `ButtonGroup`. (`undefined === undefined`)

To make this clearer, I can update the code to this:

```jsx
	const isWithinButtonGroup = !!buttonGroup;
	const isActiveWithinButtonGroup =
		isWithinButtonGroup && buttonGroup?.state === otherProps.value;
```